CAM build issue on weekly 20260513 with opensuse

### Area / Workbench affected

CAM

### Problem description

on opensuse with the last weekly snapshot we are now seeing a build error 
```
[  735s] /home/abuild/rpmbuild/BUILD/FreeCAD-weekly-20260513-build/FreeCAD-weekly-20260513/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp:570:36: warning: ‘nearDistance’ may be used uninitialized [-Wmaybe-uninitialized]
[  735s]   570 |         && height == mCameraHeight && nearDistance == mCameraNearDistance
[  735s]       |                                    ^
[  735s] /home/abuild/rpmbuild/BUILD/FreeCAD-weekly-20260513-build/FreeCAD-weekly-20260513/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp:551:11: note: ‘nearDistance’ was declared here
[  735s]   551 |     float nearDistance;
[  735s]       |           ^
[  735s] /home/abuild/rpmbuild/BUILD/FreeCAD-weekly-20260513-build/FreeCAD-weekly-20260513/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp:571:9: warning: ‘farDistance’ may be used uninitialized [-Wmaybe-uninitialized]
[  735s]   571 |         && farDistance == mCameraFarDistance) {
[  735s]       |         ^
[  735s] /home/abuild/rpmbuild/BUILD/FreeCAD-weekly-20260513-build/FreeCAD-weekly-20260513/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp:552:11: note: ‘farDistance’ was declared here
[  735s]   552 |     float farDistance;
[  735s]       |           ^
```

https://github.com/FreeCAD/FreeCAD/blob/751be50f8783ea8ac39b811c90449db8cdd2ec30/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp#L551

setting = 0.0 does let it compile

full log for reference
[weekly-build-failure-0513.txt](https://github.com/user-attachments/files/27810660/weekly-build-failure-0513.txt)

### Impact

of course it breaks if not fixed, but i seem to recall that uninitialized variables are not all treated the same for all compilers so this would also give random results at times due to that.

this is of course ignoring that there is a comment https://github.com/FreeCAD/FreeCAD/blob/751be50f8783ea8ac39b811c90449db8cdd2ec30/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp#L540 
stating there is a bug elsewhere causing this work around.

### Suggested improvement (optional)

_No response_

### References / evidence

_No response_

### Development version (if relevant)

```shell
OS: openSUSE Tumbleweed (XFCE/xfce/xcb)
Architecture: x86_64
Version: 1.2.0dev.20260512 (Git shallow)
Build date: 2026/05/12 17:54:42
Build type: Release
Branch: grafted,grafted
Hash: 758570d19c79cc4cf8b1e198b5568e7263e8e6d8
Python 3.13.13, Qt 6.11.0, Coin 4.0.3, Vtk 9.6.1, boost 1_91, Eigen3 5.0.1, PySide 6.11.0
shiboken 6.11.0, SMESH 7.7.1.0, xerces-c 3.3.0, OCC 7.9.3
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Dark/
Navigation Style/Orbit Style/Rotation Mode: Gesture/Rounded Arcball/Window center
Logical DPI/Physical DPI/Pixel Ratio: 96/81.8077/1
```

--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -226,7 +226,13 @@
         "inherits": [
           "debug",
           "conda"
-        ]
+        ],
+        "cacheVariables": {
+            "FREECAD_WARN_ERROR": {
+            "type": "BOOL",
+            "value": "ON"
+          }
+        }
       },
       {
         "name": "conda-release",

--- a/src/Base/Builder3D.cpp
+++ b/src/Base/Builder3D.cpp
@@ -1269,7 +1269,7 @@ BaseExport Vector3f stringToVector(std::string str)
     view.remove_prefix(1);
     view.remove_suffix(1);
 
-    str = view;
+    str = std::string {view};
 
     boost::char_separator<char> sep(" ,");
     boost::tokenizer<boost::char_separator<char>> tokens(str, sep);

--- a/src/Mod/CAM/App/Area.cpp
+++ b/src/Mod/CAM/App/Area.cpp
@@ -1213,21 +1213,17 @@ struct WireJoiner
 
         std::vector<VertexInfo> adjacentList;
         std::set<EdgeInfo*> edgesToVisit;
-        int count = 0;
         int skips = 0;
 
         for (auto& info : edges) {
             info.reset();
         }
 
-        int rcount = 0;
-
         for (auto& info : edges) {
             if (BRep_Tool::IsClosed(info.edge)) {
                 auto wire = BRepBuilderAPI_MakeWire(info.edge).Wire();
                 Area::showShape(wire, "closed");
                 builder.Add(comp, wire);
-                ++count;
                 continue;
             }
             gp_Pnt pt[2];
@@ -1242,7 +1238,6 @@ struct WireJoiner
                 // populate adjacent list
                 constexpr int intMax = std::numeric_limits<int>::max();
                 for (auto vit = vmap.qbegin(bgi::nearest(pt[i], intMax)); vit != vmap.qend(); ++vit) {
-                    ++rcount;
                     if (vit->pt().SquareDistance(pt[i]) > tol) {
                         break;
                     }
@@ -1376,7 +1371,6 @@ struct WireJoiner
                     }
                     Area::showShape(wire, "joined");
                     builder.Add(comp, wire);
-                    ++count;
                 }
                 break;
             }

--- a/src/Mod/CAM/PathSimulator/AppGL/DlgCAMSimulator.h
+++ b/src/Mod/CAM/PathSimulator/AppGL/DlgCAMSimulator.h
@@ -55,7 +55,7 @@ namespace CAMSimulator
 
 // use short declaration as using 'include' causes a header loop
 class MillSimulation;
-class MillSimulationState;
+struct MillSimulationState;
 struct Vertex;
 class ViewCAMSimulator;
 class GuiDisplay;

--- a/src/Mod/CAM/PathSimulator/AppGL/GuiDisplay.cpp
+++ b/src/Mod/CAM/PathSimulator/AppGL/GuiDisplay.cpp
@@ -168,7 +168,8 @@ void GuiDisplay::on_stockModelButton_clicked()
     // stock -> base -> both
     //   ^---------------'
 
-    bool sv, bv;
+    bool sv = false;
+    bool bv = false;
 
     if (stockVisible == baseVisible) {
         sv = true;

--- a/src/Mod/CAM/PathSimulator/AppGL/MillPathSegment.cpp
+++ b/src/Mod/CAM/PathSimulator/AppGL/MillPathSegment.cpp
@@ -39,11 +39,6 @@ constexpr auto pi = std::numbers::pi_v<float>;
 #define PY 1
 #define PZ 2
 
-// Maximum ratio of radius to chord length for treating arc as curved
-// Ratios above this indicate the arc is essentially a straight line
-// and should be treated as linear to avoid numerical precision issues
-constexpr float ARC_LINEARIZATION_THRESHOLD = 100000.0f;
-
 namespace CAMSimulator
 {
 

--- a/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp
+++ b/src/Mod/CAM/PathSimulator/AppGL/SimDisplay.cpp
@@ -548,8 +548,8 @@ void SimDisplay::UpdateCameraProjection(const SoCamera& camera)
 
 #else
 
-    float nearDistance;
-    float farDistance;
+    float nearDistance = 0.0;
+    float farDistance = 0.0;
 
 #endif
 

--- a/src/Mod/Measure/Gui/TaskMassProperties.cpp
+++ b/src/Mod/Measure/Gui/TaskMassProperties.cpp
@@ -1146,7 +1146,7 @@ void TaskMassProperties::tryUpdate()
         && referenceDatum->isDerivedFrom<App::Line>();
 
     const auto infoSnapshot = currentInfo;
-    QTimer::singleShot(0, this, [this, infoSnapshot, hasAxisSelection]() {
+    QTimer::singleShot(0, this, [infoSnapshot, hasAxisSelection]() {
         App::Document* doc = App::GetApplication().getActiveDocument();
         if (!doc) {
             return;

--- a/src/Mod/Measure/Gui/TaskMassProperties.h
+++ b/src/Mod/Measure/Gui/TaskMassProperties.h
@@ -57,7 +57,7 @@ public:
     bool accept() override;
     bool reject() override;
 
-    void onSelectionChanged(const Gui::SelectionChanges& msg);
+    void onSelectionChanged(const Gui::SelectionChanges& msg) override;
     void update(const Gui::SelectionChanges& msg);
     void tryUpdate();
 

--- a/src/Mod/Sketcher/Gui/TaskDlgEditSketch.h
+++ b/src/Mod/Sketcher/Gui/TaskDlgEditSketch.h
@@ -64,7 +64,7 @@ public:
     bool accept() override;
     /// is called by the framework if the dialog is rejected (Cancel)
     bool reject() override;
-    void deactivate();
+    void deactivate() override;
     bool isAllowedAlterDocument() const override
     {
         return false;

--- a/src/Mod/Spreadsheet/App/Cell.cpp
+++ b/src/Mod/Spreadsheet/App/Cell.cpp
@@ -374,7 +374,7 @@ void Cell::setContent(const char* value)
                             }
                         }
                     }
-                    else if (const auto number = freecad_cast<NumberExpression*>(parsedExpr.get())) {
+                    else if (freecad_cast<NumberExpression*>(parsedExpr.get())) {
                         // NumbersExpressions can accept more than can be parsed with strtod.
                         //   Example: 12.34 and 12,34 are both valid NumberExpressions
                         newExpr = std::move(parsedExpr);

